pkg/controller/agent/controller_test.go

```diff
@@ -149,7 +149,7 @@ func TestReconcileAgent_Reconcile(t *testing.T) {
 					ObjectMeta: metav1.ObjectMeta{
 						Name:      "testAgent-agent",
 						Namespace: "test",
-						Labels:    addLabel(defaultLabels, hash.TemplateHashLabelName, "3369118503"),
+						Labels:    addLabel(defaultLabels, hash.TemplateHashLabelName, "4210085657"),
```

Comment:
I think it's because the controller-runtime upgrade also upgrades k8s from 0.24.4 to 0.25 where PodSpec has a new  `HostUsers *bool` field.

Reply:
Nothing abnormal here, this is the normal behaviour of `hash.HashObject` and this is harmless. I verified and when the label of the Deployment is updated, Pods are left intact as their Spec didn't change.